test/object-store/sync/flx_sync.cpp

```diff
@@ -4927,6 +4927,255 @@ TEST_CASE("flx: nested collections in mixed", "[sync][flx][baas]") {
+
+    size_t num_emps = 5000;
+    size_t num_mgrs = 200;
+    size_t num_dirs = 25;
+    size_t num_total = num_emps + num_mgrs + num_dirs;
+    auto fill_person_schema = [num_emps, num_mgrs, num_dirs](SharedRealm realm) {
+        size_t cnt = 1;
+        {
+            CppContext c(realm);
+            for (size_t i = 0; i < num_mgrs; ++i) {
+                auto obj = Object::create(c, realm, "Person",
+                                          std::any(AnyDict{
+                                              {"_id", ObjectId::gen()},
+                                              {"age", static_cast<int64_t>(40 + i)},
+                                              {"role", "manager"s},
+                                              {"firstName", util::format("firstname-%1", cnt)},
+                                              {"lastName", util::format("lastname-%1", cnt)},
+                                          }));
+                ++cnt;
+            }
+        }
+        {
+            CppContext c(realm);
+            for (size_t i = 0; i < num_emps; ++i) {
+                auto obj = Object::create(c, realm, "Person",
+                                          std::any(AnyDict{{"_id", ObjectId::gen()},
+                                                           {"age", static_cast<int64_t>(20 + i % 30)},
+                                                           {"role", "employee"s},
+                                                           {"firstName", util::format("firstname-%1", cnt)},
+                                                           {"lastName", util::format("lastname-%1", cnt)}}));
+                ++cnt;
+            }
+        }
+        {
+            CppContext c(realm);
+            for (size_t i = 0; i < num_dirs; ++i) {
+                auto obj = Object::create(c, realm, "Person",
+                                          std::any(AnyDict{{"_id", ObjectId::gen()},
+                                                           {"age", static_cast<int64_t>(45 + i)},
+                                                           {"role", "director"s},
+                                                           {"firstName", util::format("firstname-9999%1", cnt)},
+                                                           {"lastName", util::format("lastname-9999%1", cnt)}}));
+                ++cnt;
+            }
+        }
+    };
+
+    enum TestState { initial, disconnected, connecting, connected, bs_downloading, bs_downloaded, bs_complete };
+    TestingStateMachine<TestState> machina(initial);
+
+    auto logger = util::Logger::get_default_logger();
+    FLXSyncTestHarness harness("flx_role_change_bootstrap", {person_schema, {"role", "firstName", "lastName"}});
+    auto& app_session = harness.session().app_session();
+    // Enable the role change bootstraps
+    // REQUIRE(app_session.admin_api.set_feature_flag(app_session.server_app_id, "allow_permissions_bootstrap",
```

Comment:
Maybe? - right now, the feature is enabled for v14 on the branch I am using, but it may go back to the feature flag once that branch is merged into master. 
These will be no longer needed, and therefore removed, once the feature branch is ready to merge into master.